### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the source of truth for this Elixir
library and leverage on latest features of ExDoc.

### CODE REVIEW

**Comment by Ziinc** [comment]
@kianmeng looks like theres 4 diff types of changes:
1. license file changes + addition to readme
2. doc badge linking
3. test file adjustments
4. mix file adjustments

can we split up this pr into two PRs. I can merge in 2,3,4. But pt1 is up to @oltarasenko

**Comment by Ziinc** [comment]
Also, could you resolve conflicts for the quickstart changes? there has been some changes in the quickstart spider example.

**Comment by oltarasenko** [comment]
> @kianmeng looks like theres 4 diff types of changes:
> 
> 1. license file changes + addition to readme
> 2. doc badge linking
> 3. test file adjustments
> 4. mix file adjustments
> 
> can we split up this pr into two PRs. I can merge in 2,3,4. But pt1 is up to @oltarasenko

I am not quite sure why do we need to remove something from the licence? Could you explain?

**Comment by kianmeng** [comment]
> Also, could you resolve conflicts for the quickstart changes? there has been some changes in the quickstart spider example.

@Ziinc Done. Please check.

**Comment by kianmeng** [comment]
> > @kianmeng looks like theres 4 diff types of changes:
> > 
> > 1. license file changes + addition to readme
> > 2. doc badge linking
> > 3. test file adjustments
> > 4. mix file adjustments
> > 
> > can we split up this pr into two PRs. I can merge in 2,3,4. But pt1 is up to @oltarasenko
> 
> I am not quite sure why do we need to remove something from the licence? Could you explain?

@oltarasenko @Ziinc 
Just remove APPENDIX from the licence, which is not part of the license terms and agreements.

File: mix.exs [hunk 1]
```diff
@@ -68,7 +52,7 @@ defmodule Crawly.Mixfile do
       {:gollum, "~> 0.4.0", hex: :new_gollum},
       {:plug_cowboy, "~> 2.0"},
       {:credo, "~> 1.5.0", only: [:dev, :test], runtime: false},
-      {:ex_doc, "~> 0.19", only: :dev, runtime: false},
+      {:ex_doc, ">= 0.0.0", only: :dev, runtime: false},
```
Comment (Ziinc) [inline mix.exs:55]: any reason why this change is necessary?
It would be preferable to leave as is to prevent major ver api changes from breaking docs generated.

File: mix.exs [hunk 2]
```diff
@@ -1,39 +1,25 @@
...
-      elixirc_options: [warnings_as_errors: true],
       deps: deps(),
-      aliases: aliases()
+      elixirc_options: [warnings_as_errors: true]
     ]
   end
 
-  defp aliases do
-    [
-      generate_documentation: &generate_documentation/1
```
Comment (Ziinc) [inline mix.exs:27]: any reason why the aliases is removed? it is used for doc gen.

**Comment by oltarasenko** [comment]
> > > @kianmeng looks like theres 4 diff types of changes:
> > > 
> > > 1. license file changes + addition to readme
> > > 2. doc badge linking
> > > 3. test file adjustments
> > > 4. mix file adjustments
> > > 
> > > can we split up this pr into two PRs. I can merge in 2,3,4. But pt1 is up to @oltarasenko
> > 
> > 
> > I am not quite sure why do we need to remove something from the licence? Could you explain?
> 
> @oltarasenko @Ziinc Just remove APPENDIX from the licence, which is not part of the license terms and agreements.

Ok, I don't have objections in this case I think!

**Comment by kianmeng** [comment]
@Ziinc I can't seem to reply to your comment, so I put everything here.

> any reason why this change is necessary?
> It would be preferable to leave as is to prevent major ver api changes from breaking docs generated.

Use the latest ExDoc to generate the documentation to leverage its latest features.

> any reason why the aliases is removed? it is used for doc gen.

This is similar to using the `assets` option for Exdoc:

```diff
   defp docs do
     [
-      source_ref: "v#{@version}",
+      assets: "documentation/assets",
```

See https://hexdocs.pm/ex_doc/Mix.Tasks.Docs.html#module-configuration

> :assets - Path to a directory that will be copied as is to the "assets" directory in the output path. Its entries may be referenced in your docs under "assets/ASSET.EXTENSION"; defaults to no assets directory.

**Comment by kianmeng** [comment]
@Ziinc @oltarasenko Any updates on this?

**Review by Ziinc — DISMISSED** [review]

**Comment by Ziinc** [comment]
@kianmeng merge requires all checks to pass, i'll fix the checks when I'm free. i'll try to get this merged in this week.

**Comment by oltarasenko** [comment]
@kianmeng Could you please run `mix format` and send it back, so I can merge it?

**Comment by kianmeng** [comment]
@oltarasenko Done. Please check.

**Comment by kianmeng** [comment]
:tada: :tada: :tada: :tada: :tada:
